### PR TITLE
Reorganize the textile simulator UI.

### PR DESCRIPTION
**Pull request by n1k0**
No user story but the basic idea is to move fields around in the UI for better general UX.

### CODE REVIEW

File: src/Page/Textile.elm [hunk 1]
```diff
@@ -754,17 +754,17 @@ productCategoryField { products } query =
...
-            , class "col-sm-4 col-form-label text-truncate"
-            ]
+    div [ class "d-flex flex-column" ]
+        [ label [ for "selector-product", class "form-label text-truncate" ]
             [ text "Catégorie" ]
         , button
-            [ class "col-sm-8 flex-fill form-select ElementSelector text-start w-auto"
-            , id "selector-product"
+            [ id "selector-product"
+            , class "form-select ElementSelector text-start w-auto"
```
Comment (vjousse) [inline src/Page/Textile.elm:767]: Just for my understanding, this TitleCase (`ElementSelector`) looks strange from the outside, why not use the same style with `-` than the other classes?

**Review by vjousse — APPROVED** [review]
LGTM :+1: Just a little question, we don't show "Teinture sur" anymore in non advanced mode but we did before. Is it the expected behavior?

File: src/Page/Textile.elm (hunk 1 above)
Comment (n1k0) [reply src/Page/Textile.elm:767]: It's the ongoing convention for custom elements in styles.scss

**Comment by n1k0** [comment]
> we don't show "Teinture sur" anymore in non advanced mode but we did before. Is it the expected behavior?

Yes, [it's a requested bug fix](https://www.notion.so/Introduire-un-champ-autonome-cat-gorie-fc0e802f1fb149e7a1699b64343defcb?pvs=4#6066295c477a4ad3aebb39ac08dabb21)

**Comment by n1k0** [comment]
I'll hold on landing this patch because Pascal has discovered [a limitation in the way we reset the fields](https://www.notion.so/Introduire-un-champ-autonome-cat-gorie-fc0e802f1fb149e7a1699b64343defcb?pvs=4#c0063204904842abbbfa2d8013e495c4) (I still have to understand what he means precisely haha)

**Comment by n1k0** [comment]
@vjousse would you mind reviewing the latest changes? The `fabricProcess` is no more a required parameter of the textile query, so we can more easily fallback on the default product fabric process when switching product category. Sorry if that sounds complicated, because it is 😅

File: public/data/textile/examples.json
Comment (vjousse) [inline public/data/textile/examples.json:1]: If it's optional (and not removed) shouldn't we keep some examples with the "fabricProcess" parameter set?

File: openapi.yaml
Comment (vjousse) [inline openapi.yaml:1]: Is this file automatically generated?

File: tests/Server/RouteTest.elm
Comment (vjousse) [inline tests/Server/RouteTest.elm:1]: I may miss something, but it looks like we don't test that removing the parameter works as expected in the generated output. Is it expected?

**Review by vjousse — APPROVED** [review]
Works as expected :+1: 
Just some nits regarding the test for my personal understanding.

File: public/data/textile/examples.json
Comment (n1k0) [reply public/data/textile/examples.json:1]: At some point we'll have to allow POs to edit example, which has been a long time requested feature. There was previously no custom fabric process used in examples as they all featured the default one associated with their category. Also we should not update examples in production before we announce a new version as discussed today.

File: openapi.yaml
Comment (n1k0) [reply openapi.yaml:1]: Nope, manually creafted and maintained with love since the very beginning. (Help).

File: tests/Server/RouteTest.elm
Comment (n1k0) [reply tests/Server/RouteTest.elm:1]: Indeed I'll add that.

**Pull request merged**